### PR TITLE
Fix Debian vulnerability matching by source package

### LINKED ISSUE #800: os vulnerability detected by 0.8.2 not in 0.9.0
@zburstein setup 0.8.2 and 0.9.0 anchore engines with the same preload db and disabled feed sync. We observed the following vulnerability for ngnix:latest and node:latest in 0.8.2 but not in 0.9.0 

```
{
  "vuln": "CVE-2017-9937",
  "feed": "vulnerabilities",
  "feed_group": "debian:10",
  "fix": "None",,
  "package": "libjbig0-2.1-3.1+b2",
  "package_name": "libjbig0",
  "package_path": "pkgdb",
  "package_type": "dpkg",
  "package_version": "2.1-3.1+b2",
  "severity": "Negligible",
  "url": "https://security-tracker.debian.org/tracker/CVE-2017-9937",
   ...
}
```

fixed artifact for CVE-2017-9937 is the same in both versions
```
postgres=# select * from feed_data_vulnerabilities_fixed_artifacts where vulnerability_id='CVE-2017-9937' and namespace_name='debian:10';
 vulnerability_id | namespace_name |  name   | version | version_format | epochless_version | include_later_versions | vendor_no_advisory | fix_metadata |         created_at         |         updated_at         | fix_observed_at
------------------+----------------+---------+---------+----------------+-------------------+------------------------+--------------------+--------------+----------------------------+----------------------------+-----------------
 CVE-2017-9937    | debian:10      | jbigkit | None    | dpkg           | None              | t                      | f                  |              | 2020-03-27 23:00:13.662518 | 2020-03-27 23:00:13.662524 |
(1 row)
```

the next thing to check was image_packages table. a vulnerability can be matched using package or the source package names

0.8.2
```
postgres=# select * from image_packages where image_id='ae2feff98a0cc5095d97c6c283dcd33090770c76d63877caa99aefbbe4343bdd' and name='libjbig0';
                             image_id                             | image_user_id |   name   |  version   | pkg_type | arch  | pkg_path | pkg_path_hash | distro_name | distro_version | like_distro | fullversion | release |                           origin                            |     src_pkg     | normalized_src_pkg | metadata_json | license | size  |         created_at         |         updated_at
------------------------------------------------------------------+---------------+----------+------------+----------+-------+----------+---------------+-------------+----------------+-------------+-------------+---------+-------------------------------------------------------------+-----------------+--------------------+---------------+---------+-------+----------------------------+----------------------------
 ae2feff98a0cc5095d97c6c283dcd33090770c76d63877caa99aefbbe4343bdd | admin         | libjbig0 | 2.1-3.1+b2 | dpkg     | amd64 | pkgdb    |               | debian      | 10             | debian      | 2.1-3.1+b2  | N/A     | Michael van der Kolff <mvanderkolff@gmail.com> (maintainer) | jbigkit-2.1-3.1 | jbigkit            |               | GPL-2+  | 78000 | 2020-12-17 20:45:34.528498 | 2020-12-17 20:45:34.528505
(1 row)
```

0.9.0
```
postgres=# select * from image_packages where image_id='ae2feff98a0cc5095d97c6c283dcd33090770c76d63877caa99aefbbe4343bdd' and name='libjbig0';
                             image_id                             | image_user_id |   name   |  version   | pkg_type | arch  | pkg_path | pkg_path_hash | distro_name | distro_version | like_distro | fullversion | release |                           origin                            |      src_pkg       | normalized_src_pkg | metadata_json | license | size  |         created_at         |         updated_at
------------------------------------------------------------------+---------------+----------+------------+----------+-------+----------+---------------+-------------+----------------+-------------+-------------+---------+-------------------------------------------------------------+--------------------+--------------------+---------------+---------+-------+----------------------------+----------------------------
 ae2feff98a0cc5095d97c6c283dcd33090770c76d63877caa99aefbbe4343bdd | admin         | libjbig0 | 2.1-3.1+b2 | dpkg     | amd64 | pkgdb    |               | debian      | 10             | debian      | 2.1-3.1+b2  | N/A     | Michael van der Kolff <mvanderkolff@gmail.com> (maintainer) | jbigkit-2.1-3.1+b2 | jbigkit+b2         |               | GPL-2+ | 78000 | 2020-12-18 01:05:42.098279 | 2020-12-18 01:05:42.098311
(1 row)
```

Notice the `normalized_src_pkg` in both cases
0.8.2 - jbigkit
0.9.0 - jbigkit+b2

Normalized source package name doesn't match the fixed artifact in the case of 0.9.0, hence the vulnerability is not matched. What still needs to be figured out is why the normalized source package is different in 0.9.0. A quick search took us [here](https://github.com/anchore/anchore-engine/blob/master/anchore_engine/services/policy_engine/engine/loaders.py#L522) but that logic hasn't changed recently and it's not yet obvious to me what upstream change could have interrupted this flow

### PR DESCRIPTION
**Pull request by wagoodman**
- Uses the `sourceVersion` in syft v0.11.0 for debian packages during vulnerability matching with the source package
- Since the `topLevelPackages` field is optional in syft output, changes the behavior to fallback to the package name in this case
- bump syft to v0.11.0

Closes #800

### CODE REVIEW

**Review by luhring — APPROVED** [review]

**Review by nightfurys — COMMENTED** [review]
lgtm, just an fyi that I don't understand the data model of the integration very well. so can't account for corner cases and such

**Comment by zhill** [comment]
I'll take a look at the test failures, I suspect we just need to update the tests.

**Comment by wagoodman** [comment]
@zhill I'm taking a look now, I see what needs to get updated

File: anchore_engine/analyzers/syft/handlers/debian.py [hunk 1]
```diff
@@ -40,12 +40,17 @@ def _all_package_info(findings, artifact):
...
+    # Normalize this for downstream consumption etc. Eventually we want to leave it split out, but for now needs a join
+    if source and source_version:
+        source = source + "-" + source_version
+    elif source:
+        source = source + "-" + version
     else:
         source = "N/A"
 
-    license = dig(artifact, "licenses") or dig(artifact, "license")
+    license = dig(artifact, "licenses")
```
Comment (zhill) [inline anchore_engine/analyzers/syft/handlers/debian.py:53]: I assume this isn't strictly part of the debian issue, but addresses the license issue

File: tests/functional/clients/standalone/package_list/fixtures/alpine.py [hunk 2]
```diff
@@ -55,8 +55,8 @@
         "type": "APKG",
         "name": "alpine-baselayout",
         "cpes": [
-            "cpe:2.3:a:*:alpine-baselayout:3.2.0-r6:*:*:*:*:*:*:*",
             "cpe:2.3:a:alpine-baselayout:alpine-baselayout:3.2.0-r6:*:*:*:*:*:*:*",
+            "cpe:2.3:a:*:alpine-baselayout:3.2.0-r6:*:*:*:*:*:*:*",
```
Comment (zhill) [inline tests/functional/clients/standalone/package_list/fixtures/alpine.py:59]: Are these tests/fixtures order sensitive?

**Review by zhill — APPROVED** [review]
lgtm, just a couple of non-blocking questions I'll go ahead and merge.

File: anchore_engine/analyzers/syft/handlers/debian.py (hunk 1 above)
Comment (wagoodman) [reply anchore_engine/analyzers/syft/handlers/debian.py:53]: This was an unrelated change. The syft source should only be pulling from "licenses" only, just noticed it while in the area was all.

File: tests/functional/clients/standalone/package_list/fixtures/alpine.py (hunk 2 above)
Comment (wagoodman) [reply tests/functional/clients/standalone/package_list/fixtures/alpine.py:59]: they are, which is important for the matching process in grype (we should be trying the most specific CPE first). We could make the order agnostic, but I think this will be more important to assert order in future releases.
